pkg/cache/clusterqueue_snapshot.go

```diff
@@ -221,3 +223,13 @@ func (c *ClusterQueueSnapshot) FindTopologyAssignmentsForWorkload(
 func (c *ClusterQueueSnapshot) IsTASOnly() bool {
 	return c.tasOnly
 }
+
+func (c *ClusterQueueSnapshot) PathToRoot() iter.Seq[*CohortSnapshot] {
```

Comment:
how about making these all functions, and putting in a single file?

Reply:
I think that making them functions is a bit clunky since it requires a naming convention like:
```
CQSnapshotPathToRoot(c *ClusterQueueSnapshot)
CohortPathToRoot(c *cohort)
```
I think leaving them as methods on the object makes more sense

Reply:
Sounds good to me